apps/aesophia_http/src/aesophia_http_handler.erl

```
@@ -253,15 +268,22 @@ parse_type(BinaryString) ->
             {error, unicode:characters_to_binary(atom_to_list(ErrorAtom))}
     end.
 
-decode_calldata_bytecode(Calldata, SerialBytecode) ->
+decode_calldata_bytecode(Calldata, SerialBytecode, BackendBin) ->
+    Backend = binary_to_atom(BackendBin, utf8),
     case deserialize(SerialBytecode) of
-        {ok, #{type_info := TypeInfo}} ->
-            decode_calldata_bytecode_(Calldata, TypeInfo);
+        %% Try a bit to be clever, if backend is not set - do some
+        %% rudimentary auto detection.
+        {ok, #{type_info := [], byte_code := Bytecode}} when Backend == default ->
+            decode_calldata_bytecode_(fate, Calldata, Bytecode);
+        {ok, #{type_info := TypeInfo}} when Backend == default; Backend == aevm ->
+            decode_calldata_bytecode_(aevm, Calldata, TypeInfo);
+        {ok, #{byte_code := Bytecode}} ->
+            decode_calldata_bytecode_(fate, Calldata, Bytecode);
```

Same here. Better to fail on unknown backend than to default to fate.